### PR TITLE
Event Locations Enhancements

### LINKED ISSUE #6402: OnIowa event locations
# Story

As an OnIowa website owner, I would like to be able to reuse locations for events happening during the same year as well as from year to year so that we do not have to enter the same location information each time.

# Additional Information

They would like the ability to have a location title/name for places like Hubbard Park
They would like the ability to store lat/long coordinates for a location.
They would like the ability to provide additional direction information for a location.
The location information will need to be made available in the API consumed by the Hawktools app in a future slice.

# Proposed Solution

- Add replicate functionality to SiteNow events (local drupal events)
- Update company name label to location name
- Add lat/long geofields
- Add additional directions field
- Create an internally focused display that shows existing events on their site with the location where they can easily find and get to an appropriate event to replicate

### PR DESCRIPTION
**Pull request by joewhitsitt**
Resolves https://github.com/uiowa/uiowa/issues/6402

Same as https://github.com/uiowa/uiowa/pull/6511 but without autofill 
stuff.

- New fields and replicate in the default event feature.
- Events backend overview for OnIowa only.

Screenshot 2023-06-19 at 9 34 50 AM

# Test

```
ddev composer install && ddev blt ds --site=oniowa.uiowa.edu && ddev drush @oniowa.local config:import --source ../config/sites/oniowa.uiowa.edu --partial -y && ddev drush @oniowa.local config-split:activate site -y && ddev drush @oniowa.local uli node/add/event
```
- Create an event node with an address location including the ~company~ location name field 👀, geolocation and driving directions. geolocation should not display on full node.
- Go to admin/content. You should see a new "Events" link next to Overview and Moderated content. That view should have basic event information displayed that the customer can use to identify similar locations for use with replicate.
- Replicate? Yeah. Try it on the event you just created.
- Check admissions/iswf for clean imports. Clean imports should mean they have the new fields and are ready to go. Those sites should also have replicate available due to the update hook.
```
ddev blt ds --site=admissions.uiowa.edu && ddev blt ds --site=iowasummerwritingfestival.uiowa.edu
```

### CODE REVIEW

**Comment by joewhitsitt** [comment]
Looks like I messed the composer merge though...

Update: fixed. Replaced the composer.* files with main and re-required geofield.

**Comment by pyrello** [comment]
Going to test this.

File: docroot/profiles/custom/sitenow/sitenow.install [hunk 1]
```diff
@@ -2459,3 +2459,16 @@ function sitenow_update_9062(&$sandbox) {
...
+ */
+function sitenow_update_9063() {
+  $config_factory = \Drupal::configFactory();
+  $config = $config_factory->getEditable('uiowa_core.settings');
+  $replicate_allowed = $config->get('uiowa_core.replicate_allowed');
+  $replicate_allowed[] = 'event';
+  $config
+    ->set('uiowa_core.replicate_allowed', $replicate_allowed)
+    ->save();
+}
```
Comment (pyrello) [inline docroot/profiles/custom/sitenow/sitenow.install:2474]: Should this be limited to just sites that have the event feature turned on?

**Comment by pyrello** [comment]
This looks good according to my testing. Just waiting for response on the outstanding question and the ready to approve.

File: docroot/profiles/custom/sitenow/sitenow.install (hunk 1 above)
Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/sitenow.install:2474]: Good question. I originally went with the minimal slice for this thinking event feature might eventually be part of the default. Hopeful thinking I guess.

I think to properly do it the other way would entail:

- a config split check on the existing update hook
- an install hook for the event feature (ignored config)
- removing it from default config
- an uninstall hook for sites that turn off the feature.

All of which can be done, just wondering if it is worth it.

Comment (pyrello) [reply docroot/profiles/custom/sitenow/sitenow.install:2474]: Strictly from a standpoint of "Do we need to do this for it to work?" its not necessary. My concern is more that we are setting up config that doesn't match the actual options for some sites. This is probably very unlikely to cause any issues. But it is also a small detail that we are likely to miss when troubleshooting some future issue that comes down to the config option being overly broad. Maybe us having this conversation is enough.

**Review by pyrello — APPROVED** [review]
Works well.

File: docroot/profiles/custom/sitenow/sitenow.install (hunk 1 above)
Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/sitenow.install:2474]: Sounds good. I appreciate you bringing this up.

oh the other gotcha is we don't have a dedicated module for the feature... https://github.com/uiowa/uiowa/issues/3458 the install hook would require a separate module to be turned on to perform this configuration change or we would have to add a drush config set to the feature instructions.

**Pull request merged**